lib/tasks/rename_duplicates_protocols.rake

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+namespace :protocols do
+  desc 'Rename dupluicates protocols '
+  task rename_duplicates_team_protocols: :environment do
+    # checking for public protocols
+    public_protocols = Protocol.select(:name, :team_id)
+                               .where.not(name: nil).where(protocol_type: 3)
+                               .group(:name, :team_id, :protocol_type)
+                               .having('COUNT(*) > 1')
+    public_protocols.each do |dup_name|
+      protocols_to_update = Protocol.where(
+        name: dup_name.name,
+        team_id: dup_name.team_id,
+        protocol_type: 3
+      )
+      protocols_to_update.each_with_index do |protocol, index|
+        next if index.zero?
```

Comment:
Just double-checking, is it really okay to assume that the first protocol (`index.zero?`) is the `dup_name` protocol? Basically, we're relying on the default ordering being the same throughout these calls. Would it be safer to also enforce ordering on all `each`, `each_with_index` calls?

Reply:
Add `.order(created_at: :asc)` to each `each_with_index`

Reply:
I've just realized that it doesn't really matter 😲.

I mean, we might _not_ always keep the first created protocol with non-renamed name, but that is still totally acceptable, we are only making sure our database state is clean.

So I think this one can be merged 🌈. We don't need to write `rspec` tests for this I believe.